server/app/models/certificate.rb

```diff
@@ -22,6 +25,27 @@ def to_path
     "#{self.grid.name}/#{self.subject}"
   end
 
+  # @return [String] with trailing newline
+  def certificate
+    pem = super
+    pem += "\n" unless pem.end_with? "\n"
```

Comment:
TBH not entirely sure if we really need these workarounds to declare #3171 fixed: https://github.com/kontena/kontena/issues/3171#issuecomment-355297618

This extra complexity here is probably useless in practice?